test/test_torch.py

```diff
@@ -12698,6 +12711,133 @@ def test_function_unwrap_message(self):
         self.assertRaisesRegex(RuntimeError, ' call to _th_lt',
                                lambda: torch.ones(1, dtype=torch.float) < torch.ones(1, dtype=torch.double))
 
+    @staticmethod
+    def _test_unary_op_overlap(self, sz, data, op):
+        # c is identical to a:
+        c = data[0:sz]
+        a = data[0:sz]
+        c_exp = torch.empty_like(c)
+        op(a, out=c_exp)
+        op(a, out=c)
+        self.assertEqual(c_exp, c)
+
+        # c and a are independent:
+        c = data[0:sz]
+        a = data[sz:2 * sz]
+        c_exp = torch.empty_like(c)
+        op(a, out=c_exp)
+        op(a, out=c)
+        self.assertEqual(c_exp, c)
+
+        # c partially overlaps with a:
+        c = data[0:sz]
+        a = data[1:sz + 1]
+        with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
+            op(a, out=c)
+
+    def test_unary_op_input_output_overlap(self):
```

Comment:
Most functions that has the input-output mem overlap check should also have the internal overlap check because a lot of them are implemented via TensorIterator. In that case, I think we should merge this test with `test_inplace_unary_mem_overlap` to avoid having two lists of operations to update.

Reply:
addressed